lib/reqTemplate.js

```diff
@@ -0,0 +1,109 @@
+"use strict";
+
+var URI = require('swagger-router').URI;
+var rbUtil = require('./rbUtil');
+var TAssembly = require('tassembly');
+
+/**
+ * Creates a function that sets a value at the path
+ * or deletes if undefined values is provided.
+ */
+function _setAtPathFun(path) {
+    var ref = path.split('.').map(function(elem) {
+        return '["' + elem + '"]';
+    }).join('');
+    var code =  'if (value !== undefined) {\n' +
+                '   obj' + ref + ' = value;\n' +
+                '} else {\n' +
+                '   delete obj' + ref +';\n' +
+                '}';
+    /* jslint evil: true */
+    return new Function('obj', 'value', code);
+}
+
+function _createTemplateResolvers(subspec, reqPart, path, resolvers) {
+    var resolveTemplate = function(templateSpec, templatePath) {
```

Comment:
It seems to me this doesn't have to be a `var`, `function resolverTemplate(...)` is highly recommended